tests/neg/OrType.scala

```diff
@@ -2,7 +2,7 @@ class B(val x: Int)
 class C(val x: Double)
 
 object Test{
-  def bar(x: B | C): Int | Double = x.x
+  def bar(x: B | C): Int | Double = x.x  // error
```

Comment:
@smarter I pushed `union-types-narrow-prefix` to staging. I am a bit skeptical this one can be made to work, though. In particular I don't know what the theoretical foundations of doing it this way would be.